[RDY] vim-patch:8.0.1620,8.1.{96,340}

janlazo opened the pull request:
**vim-patch:8.0.1620: reading spell file has no good EOF detection**

Problem:    Reading spell file has no good EOF detection.
Solution:   Check for EOF at every character read for a length field.
vim/vim@e26e0d2

**vim-patch:8.1.0096: inconsistent use of the word autocommands**

Problem:    Inconsistent use of the word autocommands.
Solution:   Don't use auto-commands or "auto commands".
vim/vim@8c55533

**vim-patch:8.1.0340: no test for :spellinfo**

Problem:    No test for :spellinfo.
Solution:   Add a test. (Dominique Pelle, closes vim/vim#3394)
vim/vim@9049b68

janlazo committed:
vim-patch:8.1.0340: no test for :spellinfo

Problem:    No test for :spellinfo.
Solution:   Add a test. (Dominique Pelle, closes vim/vim#3394)
https://github.com/vim/vim/commit/9049b686121367941bf534c041975938135c7e20

janlazo committed:
oldtests: skip Test_spellinfo()

nvim supports only `set encoding=utf8`.

justinmk commented:
```
==27542==ERROR: AddressSanitizer: global-buffer-overflow on address 0x000001bbefa1 at pc 0x0000004d9202 bp 0x7ffda709f8e0 sp 0x7ffda709f090
READ of size 6 at 0x000001bbefa1 thread T0
    #0 0x4d9201 in __asan_memcpy /local/mnt/workspace/tmp/ubuntu_rel/llvm/utils/release/final/llvm.src/projects/compiler-rt/lib/asan/asan_interceptors.cc:466:3
    #1 0xfcb9d9 in xmemdupz /home/travis/build/neovim/neovim/build/../src/nvim/memory.c:215:10
    #2 0xa379eb in tv_list_append_string /home/travis/build/neovim/neovim/build/../src/nvim/eval/typval.c:553:28
    #3 0x951562 in f_spellbadword /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:15602:3
    #4 0x8305a4 in call_func /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:6372:11
```

janlazo commented:
Same error. Why is there a buffer flow?

```
==32617==ERROR: AddressSanitizer: global-buffer-overflow on address 0x000001bcb981 at pc 0x0000004d9352 bp 0x7fff288363c0 sp 0x7fff28835b70
READ of size 6 at 0x000001bcb981 thread T0
    #0 0x4d9351 in __asan_memcpy /local/mnt/workspace/tmp/ubuntu_rel/llvm/utils/release/final/llvm.src/projects/compiler-rt/lib/asan/asan_interceptors.cc:466:3
    #1 0xfd3869 in xmemdupz /home/travis/build/neovim/neovim/build/../src/nvim/memory.c:215:10
    #2 0xa3e1eb in tv_list_append_string /home/travis/build/neovim/neovim/build/../src/nvim/eval/typval.c:553:28
    #3 0x957ae2 in f_spellbadword /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:15618:3
    #4 0x8369c4 in call_func /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:6378:11
    #5 0x84a978 in get_func_tv /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:6124:11
```

janlazo committed:
vim-patch:8.0.1620: reading spell file has no good EOF detection

Problem:    Reading spell file has no good EOF detection.
Solution:   Check for EOF at every character read for a length field.
https://github.com/vim/vim/commit/e26e0d2b83c2875b9829b884c2ababf8ca771f7e

janlazo committed:
vim-patch:8.1.0096: inconsistent use of the word autocommands

Problem:    Inconsistent use of the word autocommands.
Solution:   Don't use auto-commands or "auto commands".
https://github.com/vim/vim/commit/8c55533c6f109db2a0fff69651887f9474eb09c6

janlazo committed:
lint
